src/js/components/Carousel.js

```diff
@@ -1,45 +1,174 @@
+  },
+
+  componentWillUnmount: function() {
+    clearInterval(this.slideAnimation);
+
+    window.removeEventListener('resize', this._onWindowResize);
+  },
+
+  slideAnimation: null,
+
+  _setSlideInterval: function() {
+    this.slideAnimation = setInterval(function() {
+      var activeIndex = this.state.activeIndex;
+      var numSlides = this.props.children.length;
+
+      this.setState({
+        activeIndex: (activeIndex + 1) % numSlides
+      });
+
+      if (!this.props.infinite && activeIndex === numSlides - 1) {
+        clearInterval(this.slideAnimation);
+      }
+    }.bind(this), this.props.autoplaySpeed);
   },
 
   _onSelect: function (index) {
     if (index !== this.state.activeIndex) {
       this.setState({
-        activeIndex: index,
-        priorIndex: this.state.activeIndex,
-        sequence: this.state.sequence += 1
+        activeIndex: index
+      });
+    }
+  },
+
+  _onMouseOver: function() {
+    if (this.props.autoplay) {
+      clearInterval(this.slideAnimation);
+    }
+
+    if (!this.props.persistentNav) {
+      this.setState({
+        hideControls: false
+      });
+    }
+  },
+
+  _onMouseOut: function() {
+    if (this.props.autoplay && (this.props.infinite || this.state.activeIndex !== this.props.children.length - 1)) {
+      this._setSlideInterval();
+    }
+
+    if (!this.props.persistentNav) {
+      this.setState({
+        hideControls: true
       });
     }
   },
 
-  // children should be an array of Tile
+  _onWindowResize: function () {
+    this.setState({
+      width: this.refs.carousel.offsetWidth
+    });
+  },
+
+  _slidePrev: function() {
+    var numSlides = this.props.children.length;
+    this.setState({
+      activeIndex: (this.state.activeIndex + numSlides - 1) % numSlides
+    });
+  },
+
+  _slideNext: function() {
+    var numSlides = this.props.children.length;
+    this.setState({
+      activeIndex: (this.state.activeIndex + 1) % numSlides
+    });
+  },
+
+  _renderPrevButton: function() {
+    if (this.props.infinite || this.state.activeIndex !== 0) {
+      return (
+        <svg className={CLASS_ROOT + '__arrow ' + CLASS_ROOT + '__arrow--prev'} viewBox="0 0 36 72" version="1.1"
+          onClick={this._slidePrev}>
+          <line x1="36" y1="0" x2="0" y2="36" />
+          <line x1="0" y1="36" x2="36" y2="72" />
+        </svg>
+      );
+    }
+  },
+
+  _renderNextButton: function() {
+    if (this.props.infinite || this.state.activeIndex !== this.props.children.length - 1) {
+      return (
+        <svg className={CLASS_ROOT + '__arrow ' + CLASS_ROOT + '__arrow--next'} viewBox="0 0 36 72" version="1.1"
+          onClick={this._slideNext}>
+          <line x1="0" y1="0" x2="36" y2="36" />
+          <line x1="36" y1="36" x2="0" y2="72" />
+        </svg>
+      );
+    }
+  },
+
   render: function () {
     var classes = [CLASS_ROOT];
-    classes.push(CLASS_ROOT + "--toggle-" + (this.state.sequence % 2));
     if (this.props.className) {
```

Comment:
This is extremely minor, but our convention is to put any passed in className as the last class. It makes it slightly easier to read the class names in the browser developer tools when they are grouped that way.